crates/anemo/src/crypto.rs

```diff
@@ -87,11 +94,20 @@ impl rustls::client::ServerCertVerifier for CertVerifier {
             }
             _ => return Err(rustls::Error::UnsupportedNameType),
         };
-        let expected_dns_nameref = webpki::DnsNameRef::try_from_ascii_str(&self.0)
-            .map_err(|_| rustls::Error::UnsupportedNameType)?;
-        if dns_nameref.as_ref() != expected_dns_nameref.as_ref() {
-            return Err(rustls::Error::UnsupportedNameType);
-        }
+        // Client server_name needs to match one of our server_names
+        let _ = self
```

Comment:
is the `let _ = ` needed since you are using `?`?

Reply:
removed `let _ = `